core/finality-grandpa/src/communication/mod.rs

```diff
@@ -436,7 +443,9 @@ impl<B: BlockT, N: Network<B>> NetworkBridge<B, N> {
 			format!("Failed to receive on unbounded receiver for round {}", round.0)
 		));
 
-		let incoming = incoming.select(out_rx);
+		// TODO: Do I understand correctly that we combine the incoming Grandpa messages with the messages we are
+		// sending out to other nodes and in addition thereby also send to ourself?
```

Comment:
yes! this means that the logic for importing our own votes is unified with the logic for importing others' votes in the GRANDPA crate - reducing code paths in the vote processing pipeline and making bugs less likely.